lib/ansible/modules/clustering/kubevirt/k8s_service.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2018, KubeVirt Team <@kubevirt>
+# Apache License, Version 2.0
+# (see LICENSE or http://www.apache.org/licenses/LICENSE-2.0)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: k8s_service
+
+short_description: Manage Services on Kubernetes
+
+description:
+    - Use Openshift Python SDK to manage Services on Kubernetes
+
+version_added: "2.8"
+
+author: KubeVirt Team (@kubevirt)
+
+options:
+    name:
+        description:
+            - Use to specify a Service object name.
+        required: true
+        type: str
+    namespace:
+        description:
+            - Use to specify a Service object namespace.
+        required: true
+        type: str
+    type:
+        description:
+            - Specifies the type of Service to create.
+            - https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types
+        choices:
+            - NodePort
+            - ClusterIP
+            - LoadBalancer
+            - ExternalName
+        required: true
+    ports:
+        description:
+            - A list of ports to expose.
+            - https://kubernetes.io/docs/concepts/services-networking/service/#multi-port-services
+        required: true
+        type: list
+    selector:
+        description:
+            - Label selectors identify objects this Service should apply to.
+            - https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/
+        required: true
```

Comment:
This is documented as required, but not really required if I see correctly.

Reply:
Yes, but considering that there are only very specific cases where you'd want to not have a selector, it's probably a better choice from a UX perspective to have you set it as empty if you really know you don't want it.

Reply:
I am ok whether it's required or not. What I mean is that doc says something different than the code, if I see correctly.